x/twap/strategy_test.go

```diff
@@ -6,26 +6,40 @@ import (
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 
+	"github.com/osmosis-labs/osmosis/v13/app/apptesting"
 	"github.com/osmosis-labs/osmosis/v13/app/apptesting/osmoassert"
 	"github.com/osmosis-labs/osmosis/v13/osmomath"
 	"github.com/osmosis-labs/osmosis/v13/x/twap"
 	"github.com/osmosis-labs/osmosis/v13/x/twap/types"
 )
 
+type TwapStrategyTestSuite struct {
+	apptesting.KeeperTestHelper
+}
+
+func TestTwapStrategyTestSuite(t *testing.T) {
+	suite.Run(t, new(TwapStrategyTestSuite))
+}
+
+func (suite *TwapStrategyTestSuite) SetupTest() {
+	suite.Setup()
+}
+
 // TestComputeArithmeticTwap tests computeTwap on various inputs.
 // TODO: test both arithmetic and geometric twap.
 // The test vectors are structured by setting up different start and records,
 // based on time interval, and their accumulator values.
 // Then an expected TWAP is provided in each test case, to compare against computed.
-func TestComputeTwap(t *testing.T) {
+func (suite *TwapStrategyTestSuite) TestComputeTwap() {
 	tests := map[string]computeTwapTestCase{
 		"arithmetic only, basic: spot price = 1 for one second, 0 init accumulator": {
 			startRecord: newOneSidedRecord(baseTime, sdk.ZeroDec(), true),
 			endRecord:   newOneSidedRecord(tPlusOne, OneSec, true),
 			quoteAsset:  denom0,
 			twapStrategies: []twap.TwapStrategies{
-				twap.ArithmeticTwapStrategy,
+				twap.ArithmeticTwapStrategy{suite.App.TwapKeeper},
```

Comment:
@p0mvn getting an error here that says;
```
cannot use twap.ArithmeticTwapStrategy{…} (value of type twap.arithmetic) as type twap.twapStrategies in array or slice literal:
	twap.arithmetic does not implement twap.twapStrategies (computeTwap method has pointer receiver)go
```

since `arithmetic` and `geometric` are their own type, do i have to typecast to make it a part of `twapStrategy` type?

Reply:
nvm i fixed it! :))